client/cgo/verification.go

```diff
@@ -112,8 +142,15 @@ func cgoVerifyAuthPath(treeHash unsafe.Pointer, treeHashSize C.int,
 	ap.TreeNonce = tn
 	ap.LookupIndex = li
 	ap.PrunedTree = pt
+	ap.Leaf = &leafNode{
```

Comment:
Well, if you set `ap.Leaf`, you can do,

```
-if merkletree.VerifyAuthPath(ap, leafi, leafc, uint32(leafLevel), int(isLeafEmpty) == 1, th) {
+if merkletree.VerifyAuthPath(ap, th) {
```

> Also, these two values would only be nil in only some cases, no?

Not if all you're doing with the leaf is calling `VerifyAuthPath`.  But that's sort of what I was getting at above in my "important point".  These client functions will probably evolve, as we flesh out the messages, to be called with a more complete leaf.

I'm thinking that eventually `cgoVerifySignature`, `cgoVerifyCommitment`, will be combined in more elaborate `cgoDoTheClientBusiness` type thing.